### PR TITLE
Implement go xlang wrapper for reading from Bigtable

### LINKED ISSUE #27509: [Feature Request]:[Go SDK]: BigTableIO should have scalable reads.
### What would you like to happen?

The Go BigTableIO should take advantage of BigTable providing a splitting API with the https://pkg.go.dev/cloud.google.com/go/bigtable#Table.SampleRowKeys method, in order to be authored as a Splitable Do Fn, which could then build RowSets to read from per split.

This would allow the bigtable IO to scale up it's reads.

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [X] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by steveniemitz**
This addresses #27509 by adding the scalable Java BigTableIO via Xlang.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by steveniemitz** [comment]
Run XVR_GoUsingJava_Dataflow PostCommit

**Comment by jrmccluskey** [comment]
The XVR GoUsingJava tests are known perma-red (#21645) feel free to ignore failures that aren't related to your testing

**Comment by steveniemitz** [comment]
> The XVR GoUsingJava tests are known perma-red (#21645) feel free to ignore failures that aren't related to your testing

ah ok, my new test is passing at least!

File: sdks/go/pkg/beam/core/runtime/xlangx/payload.go [hunk 1]
```diff
@@ -26,37 +26,49 @@ import (
...
+}
+
+// EncodeStructPayload takes a native Go struct and returns a marshaled
+// ExternalConfigurationPayload proto, containing a Schema representation of
+// the original type and the original value encoded as a Row. This is intended
+// to be used as the expansion payload for an External transform.
+func EncodeStructPayload(pl any) ([]byte, error) {
+	ecp, err := CreateExternalConfigurationPayload(pl)
+	if err != nil {
+		return []byte{}, nil
```
Comment (jrmccluskey) [inline sdks/go/pkg/beam/core/runtime/xlangx/payload.go:69]:
```suggestion
		return []byte{}, err
```

File: sdks/go/test/integration/io/xlang/bigtable/bigtable_test.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	project := matches[1]
+	instance := matches[2]
+
+	admin, err := bt.NewAdminClient(context.Background(), project, instance)
+	if err != nil {
+		panic(err)
+	}
+
+	table := createTempTable(admin)
+	defer deleteTempTable(admin, table)
```
Comment (jrmccluskey) [inline sdks/go/test/integration/io/xlang/bigtable/bigtable_test.go:154]: Prefer t.Cleanup over defer in test cases 

https://pkg.go.dev/testing#T.Cleanup

File: sdks/go/pkg/beam/core/runtime/xlangx/payload.go (hunk 1 above)
Comment (steveniemitz) [reply sdks/go/pkg/beam/core/runtime/xlangx/payload.go:69]: whoops good catch

File: sdks/go/test/integration/io/xlang/bigtable/bigtable_test.go (hunk 2 above)
Comment (steveniemitz) [reply sdks/go/test/integration/io/xlang/bigtable/bigtable_test.go:154]: 👍

**Review by lostluck — APPROVED** [review]
LGTM Wow, thanks!

**Pull request merged**